src/openms/include/OpenMS/QC/Ms2IdentificationRate.h

```diff
@@ -69,6 +71,17 @@ namespace OpenMS
     /// container that stores results
     std::vector<IdentificationRateData> rate_result_;
 
+    /// returns number of all ms2 spectra in an MSExperiment
+    Size getMS2Count_(const MSExperiment& exp);
+
+    /// increases a 'counter' if PeptideIdentification is considered 'target' (target/decoy annotation needed)
+    /// if force_index is true, 'counter' is increased regardless
+    /// is static so that it can be used with MapUtilities::applyFunctionOnPeptideIDs() without creating a new object for each ID
+    static void countPepID_(const PeptideIdentification& id, Size& counter, bool force_index);
```

Comment:
the interface is a bit... complicated?!
how about `static bool isTargetPeptide(PepID, count_unmatched)`
i.e. use the return value as counter and go
`counter += isTargetPeptide(...)` in calling code?